src/Umbraco.Core/StringExtensions.cs

```diff
@@ -1465,10 +1470,7 @@ internal static string ToValidXmlString(this string text)
         /// <returns></returns>
         internal static Guid ToGuid(this string text)
         {
-            var md5 = MD5.Create();
-            byte[] myStringBytes = Encoding.ASCII.GetBytes(text);
-            byte[] hash = md5.ComputeHash(myStringBytes);
-            return new Guid(hash);
+            return new Guid(text.GenerateHash());

```

Comment:
What about adding something like this?

 https://github.com/LogosBible/Logos.Utility/blob/master/src/Logos.Utility/GuidUtility.cs#L34

See discussion here:

https://stackoverflow.com/a/5657517/32908